optimade/server/entry_collections/entry_collections.py

```diff
@@ -43,22 +96,114 @@ def get_attribute_fields(self) -> set:
                 attributes = attributes[next_key]
         return set(attributes["properties"].keys())
 
-    @abstractmethod
-    def find(
-        self, params: EntryListingQueryParams
-    ) -> Tuple[List[EntryResource], int, bool, set]:
+    def handle_query_params(
+        self, params: Union[EntryListingQueryParams, SingleEntryQueryParams]
+    ) -> dict:
+        """ Parse and interpret the backend agnostic query parameter models into a dictionary
+        that can be used by the specific backend. Currently returns the pymongo
+        interpretation of the parameters, which will need modification for modified
+        for other backends.
+
+        Parameters:
+            params: the initialized query parameter model from the server.
+
+        Raises:
+            HTTPException: if too large a page limit is provided.
+            BadRequest: if an invalid request is made, e.g. with incorrect fields
+                or response format.
+
+        Returns:
+            dict: a dictionary representation of the query parameters, ready to be used
+                by pymongo.
+
         """
-        Fetches results and indicates if more data is available.
+        cursor_kwargs = {}
 
-        Also gives the total number of data available in the absence of page_limit.
+        if getattr(params, "filter", False):
+            tree = self.parser.parse(params.filter)
+            cursor_kwargs["filter"] = self.transformer.transform(tree)
+        else:
+            cursor_kwargs["filter"] = {}
 
-        Args:
-            params (EntryListingQueryParams): entry listing URL query params
+        if (
+            getattr(params, "response_format", False)
+            and params.response_format != "json"
+        ):
+            raise BadRequest(
+                status_code=400, detail="Only 'json' response_format supported"
+            )
+
+        if getattr(params, "page_limit", False):
+            limit = params.page_limit
+            if limit > CONFIG.page_limit_max:
+                raise HTTPException(
+                    status_code=403,  # 403 Forbidden is enforced by the specification
```

Comment:
You should make this a proper exception in optimade.server.exceptions and use it here. Then you only need to provide `detail` here afterwards.
Remember to update the docstring.